cli/server/server.go

```diff
@@ -209,22 +213,42 @@ func dumpDB(ctx *cli.Context) error {
 	if count == 0 {
 		count = chainCount - start
 	}
+
+	var rootsWriter *io.BinWriter
+	if out := ctx.String("state"); out != "" {
+		rootsStream, err := os.Create(out)
+		if err != nil {
+			return cli.NewExitError(fmt.Errorf("can't create file for state roots: %w", err), 1)
+		}
+		defer rootsStream.Close()
+		rootsWriter = io.NewBinWriterFromIO(rootsStream)
+	}
 	if start != 0 {
 		writer.WriteU32LE(start)
 	}
 	writer.WriteU32LE(count)
+
+	if rootsWriter != nil {
+		rootsWriter.WriteU32LE(start)
+		rootsWriter.WriteU32LE(count)
```

Comment:
Same thing here, do we really have this `count` of states when exporting whole chain we have?